DoLahTest/ParserUnitTest.cpp

```diff
@@ -379,6 +387,130 @@ namespace DoLahTest {
         }
     }
 
+    // From here, integrated test on special cases of 'toDateFormat' for relative time descriptions.
+    // The time will auto-filled, and the behaviour changes depending on the time of running.
+    // Test will be disabled in a certain condition where it cannot run properly.
+    TEST_METHOD(DateGivenDayAndMonth) {
+        if (month == 1 && day == 1) {
+            Assert::IsTrue(true);
+        } else { // test will not function in every new year
+            std::string input = "1st January";
+            std::string expected = strfyTime(0, 0, 0, 1, 1, nextYear);
+            std::string actual = tmToString(DoLah::DateTimeParser::toDateFormat(DoLah::ParserLibrary::explode(input, " ")));
+            Assert::AreEqual(expected, actual);
+        }
+    }
+
+    TEST_METHOD(DateGivenDay) {
+        if (day == 1) {
+            Assert::IsTrue(true);
+        } else { // test will not function on the 1st of every month
+            std::string input = "1st";
+            std::string expected = strfyTime(0, 0, 0, 1, nextMonth, year);
+            std::string actual = tmToString(DoLah::DateTimeParser::toDateFormat(DoLah::ParserLibrary::explode(input, " ")));
+            Assert::AreEqual(expected, actual);
+        }
+    }
+
+    TEST_METHOD(DateGivenMonthAndDay) {
+        if (month == 1 && day == 1) {
+            Assert::IsTrue(true);
+        } else { // test will not function in every new year
+            std::string input = "January 1st";
+            std::string expected = strfyTime(0, 0, 0, 1, 1, nextYear);
+            std::string actual = tmToString(DoLah::DateTimeParser::toDateFormat(DoLah::ParserLibrary::explode(input, " ")));
+            Assert::AreEqual(expected, actual);
+        }
+    }
+
+    TEST_METHOD(Tomorrow) {
+        if (remainderDays < 1) {
+            Assert::IsTrue(true);
+        } else { // test will not function on the last day of the month
+            std::string input = "tomorrow";
+            std::string expected = strfyTime(0, 0, 0, nextDay, month, year);
+            std::string actual = tmToString(DoLah::DateTimeParser::toDateFormat(DoLah::ParserLibrary::explode(input, " ")));
+            Assert::AreEqual(expected, actual);
+        }
+    }
```

Comment:
this method stop working on last day of month. it should still work regardless the current day.